pom.xml

```diff
@@ -133,25 +146,62 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-surefire-plugin</artifactId>
-                <version>3.0.0-M8</version>
+                <version>${maven-surefire-plugin.version}</version>
+            </plugin>
+
+            <!-- report-->
+            <plugin>
+                <groupId>org.owasp</groupId>
+                <artifactId>dependency-check-maven</artifactId>
+                <version>${dependency-check-maven.version}</version>
+                <configuration>
+                    <failBuildOnCVSS>5</failBuildOnCVSS>
+                </configuration>
             </plugin>
         </plugins>
     </build>
 
     <profiles>
         <profile>
-            <id>signing</id>
-            <activation>
-                <property>
-                    <name>gpg.passphrase</name>
-                </property>
-            </activation>
+            <!-- use custom release profile "sonatype-release" instead of Maven's "release" profile and keep the latter disabled to prevent plugins from interfering with each other.
+                 secrets and additional properties necessary for OSSH release are defined in settings.xml which is not included for security reasons-->
+            <id>sonatype-release</id>
             <build>
                 <plugins>
+                    <plugin>
+                        <groupId>org.apache.maven.plugins</groupId>
+                        <artifactId>maven-source-plugin</artifactId>
+                        <version>${maven-source-plugin.version}</version>
+                        <executions>
+                            <execution>
+                                <id>attach-sources</id>
+                                <phase>verify</phase>
+                                <goals>
+                                    <goal>jar-no-fork</goal>
+                                </goals>
+                            </execution>
+                        </executions>
+                    </plugin>
+                    <plugin>
+                        <groupId>org.apache.maven.plugins</groupId>
+                        <artifactId>maven-javadoc-plugin</artifactId>
```

Comment:
I wouldn't move the javadoc plugin into the `sonatype-release` profile. We don't do it yet, but in principle, we could build the javadocs as a kind of verification, too. (We might not want to notice only during a release when the javadocs are broken...) And for that, we should IMO use a consistent plugin configuration.